MatrixTest.java

```diff
@@ -0,0 +1,163 @@
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotSame;
+import org.junit.Test;
+
+
+public class MatrixTest{
+    //Matrix(double[][] A){}
+
+    //Matrix(double[][] A, int m, int n){}
+    @Test
+    public void baseCtorSameLength(){
+	double[][] array = {
+		{1,2,3},
+		{4,5,6},
+		{7,8,9}
+	};
+	Matrix m = new Matrix(array, 3, 3);
+	double[][] array2 = m.getArray();
+        assertEquals(array, array2);
+
+    }
+    @Test
+    public void baseCtorNotSameObj(){
+	double[][] array = {
+		{1,2,3},
+		{4,5,6},
+		{7,8,9}
+	};
+	Matrix m = new Matrix(array, 3, 3);
+	double[][] array2 = m.getArray();
+        assertNotSame(array, array2);
+    }
+    @Test
+    public void tooLong(){
+        double[][] array = {
+            {1,2,3},
+            {4,5,6, 6.5},
+            {7,8,9}
+        };
+        Matrix m = new Matrix(array, 3, 3);
+
+        double[][] array2 = {
+            {1,2,3},
+            {4,5,6},
+            {7,8,9}
+        };
+        assertEquals(array2, m.getArray());
+    }
+    @Test
+    public void tooShort(){
+        double[][] array = {
+            {1,2,3},
+            {4,5,},
+            {7,8,9}
+        };
+        Matrix m = new Matrix(array, 3, 3);
+        
+        //TODO is this what we want to happen when a row is too short?
```

Comment:
Yes, I think this is what should happen. We should probably also test the case where they provide a 4x4 matrix but ask for a 3x3 one.